src/VecSim/memory/vecsim_malloc.cpp

```diff
@@ -36,12 +37,9 @@ void *VecSimAllocator::reallocate(void *p, size_t size) {
         return this->allocate(size);
     }
     size_t oldSize = getPointerAllocationSize(p);
-    if (oldSize >= size) {
-        return p;
-    }
     void *new_ptr = this->allocate(size);
     if (new_ptr) {
-        memcpy(new_ptr, p, oldSize);
+        memcpy(new_ptr, p, MIN(oldSize, size));
```

Comment:
why the change here?

Reply:
because if `oldSize>size`, we don't want to copy `oldSize` bytes to the new buffer of size `size` (we'll get segmentation fault...)